internal/api/controllers/private/runsCreateV2.go

```diff
@@ -0,0 +1,225 @@
+		satellite = false
+	}
+
+	if runInput.Hosts == nil {
+		satellite = false
+		for _, host := range *runInput.Hosts {
+			if host.AnsibleHost == nil || *host.AnsibleHost == "" {
+				ansible = false
+			}
+			if host.InventoryId == nil {
+				satellite = false
+			}
+		}
+	}
+	return
+}
+
+func buildMetadata(satRequest bool, correlationId uuid.UUID, runInput RunInputV2) (metadata map[string]string) {
+	if !satRequest {
+		metadata["crc_dispatcher_correlation_id"] = correlationId.String()
+		return
+	}
+
+	var hosts string
+	for _, host := range *runInput.Hosts {
+		hosts += *host.InventoryId
+	}
+
+	principal := string(runInput.Principal)
+	var principalHash string
+
+	if principal != "" {
+		principalHash = fmt.Sprintf("%x", sha256.Sum256([]byte(principal)))
+	}
+
+	metadata = map[string]string {
+		"operation":         "run",
+		"correlation_id":    correlationId.String(),
+		"playbook_run_name": string(runInput.Name),
+		"playbook_run_url": string(runInput.Url),
+		"sat_id":	string(*runInput.RecipientConfig.SatId),
+		"sat_org_id": string(*runInput.RecipientConfig.SatOrgId),
+		"initiator_user_id": principalHash,
+		"hosts": hosts,
+	}
+	return
+}
+
+//go:generate fungen -types RunInputV2,*RunCreated -methods PMap -package private -filename utils.v2.gen.go
+func (this *controllers) ApiInternalV2RunsCreate(ctx echo.Context) error {
+	var input RunInputV2List
+
+	db.SetLog(this.database, utils.GetLogFromEcho(ctx))
+	defer db.ClearLog(this.database)
+
+	err := utils.ReadRequestBody(ctx, &input)
+	if err != nil {
+		utils.GetLogFromEcho(ctx).Error(err)
+		return ctx.NoContent(http.StatusBadRequest)
+	}
+
+	// send all requests to cloud connector concurrently
+	result := input.PMapRunCreated(func(runInput RunInputV2) *RunCreated {
+		ansibleReq, satReq := CheckReqFields(runInput)
+		if (ansibleReq && satReq) || (!ansibleReq && !satReq) {
+			return runCreateError(http.StatusBadRequest)
+		}
+
+		cloudConnectorDirective := ansibleCloudConnectorDirective
+		requestType := "ansible"
+		if satReq {
+			requestType = "satellite"
+			cloudConnectorDirective = satCloudConnectorDirective
+		}
+
+		recipient, err := uuid.Parse(string(runInput.Recipient))
+		if err != nil {
+			instrumentation.InvalidRecipientId(ctx, string(runInput.Recipient), err, requestType)
+			return runCreateError(http.StatusBadRequest)
+		}
+
+		correlationId := uuid.New()
+
+		if this.config.GetBool("demo.mode") {
+			correlationId = uuid.UUID{}
+		}
+
+		context := utils.WithRequestType(ctx.Request().Context(), requestType)
+		context = utils.WithCorrelationId(context, correlationId.String())
+		context = utils.WithAccount(context, string(runInput.OrgId))
+
+		// take from the rate limit bucket
+		this.rateLimiter.Wait(ctx.Request().Context())
+
+		metadata := buildMetadata(satReq, correlationId, runInput)
+
+		messageId, notFound, err := this.cloudConnectorClient.SendCloudConnectorRequest(
+			context,
+			string(runInput.OrgId),
```

Comment:
org_id cannot be used here. It needs to be translated to account first using https://github.com/RedHatInsights/playbook-dispatcher/blob/master/internal/api/connectors/tenants/types.go#L15